### PR TITLE
Yo Support Updates with Windows and Workspace support

### PR DESCRIPTION
**Pull request by christopherbryanboyd**

### CODE REVIEW

File: cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java [hunk 1]
```diff
@@ -299,6 +305,22 @@ protected ProjectTemplatesArgs getProjectTemplateArgs(
...
 		properties.put("setViewType", createArgs.getViewType());
 
+		WorkspaceProvider workspaceProvider = getBladeCLI().getWorkspaceProvider(new File(createArgs.getBase()));
+
+		try {
+			if (workspaceProvider != null) {
+				File workspaceLocation = workspaceProvider.getWorkspaceDir(getBladeCLI());
+
+				if (workspaceLocation != null) {
+					properties.put("setModulesLocation", _getDefaultModulesDir().toString());
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java]: Having to set these properties like this means our ProjectTemplateArgs ext api is broken.  If we have to resort to stringify these methods, then we might as well just pass hashMaps all around and remove the arg extensions api.

File: cli/src/main/resources/com/liferay/blade/cli/util/dependencies/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (gamerson) [inline cli/src/main/resources/com/liferay/blade/cli/util/dependencies/package.json]: This should all be added to the custom template not in `cli`

File: cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java [hunk 3]
```diff
@@ -155,6 +156,11 @@ else if (template.equals("modules-ext")) {
 				return;
 			}
 		}
+		else if (template.equals("js-widget")) {
+			NodeUtil.downloadNode();
```
[the remark this answers is not in the document]
Comment (christopherbryanboyd) [reply cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java]: I'll relocate this code into the js-widget template class

File: cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java (hunk 1 above)
Comment (christopherbryanboyd) [reply cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java]: Agreed that this isn't ideal, and I think eventually we can find a better solution than passing in the method names / values as properties to Project Templates.

File: cli/src/main/resources/com/liferay/blade/cli/util/dependencies/package.json (hunk 2 above)
Comment (christopherbryanboyd) [reply cli/src/main/resources/com/liferay/blade/cli/util/dependencies/package.json]: Agreed :+1:

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/gamerson/liferay-blade-cli/pull/929. [See changes here.](https://github.com/gamerson/liferay-blade-cli/compare/63e4bb87~4...63e4bb87) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**